dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -122,6 +122,46 @@ public WorkflowResource(HttpClient client, UserDAO userDAO, TokenDAO tokenDAO, W
         return workflowDAO.findAll();
     }
 
+    @GET
+    @Path("/{workflowId}/restub")
+    @Timed
+    @UnitOfWork
+    @ApiOperation(value = "Restub a workflow", notes = "Restubs a full, unpublished workflow.", response = Workflow.class)
+    public Workflow restub(@ApiParam(hidden = true) @Auth User user,
+            @ApiParam(value = "workflow ID", required = true) @PathParam("workflowId") Long workflowId) {
+        Workflow workflow = workflowDAO.findById(workflowId);
+        // Check that workflow is valid to restub
+        if (workflow.getIsPublished()) {
+            throw new CustomWebApplicationException("A workflow must be unpublished to restub.", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        if (workflow.getMode().toString().equals("STUB")) {
+            throw new CustomWebApplicationException("The given workflow is already a stub.", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        Workflow newWorkflow = new Workflow();
```

Comment:
Talked with @agduncan94  
The reasoning here is that there was difficulty in stripping associated versions and source files from the object. Creating a new one seemed like the better option, but should be improved.